staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2982,6 +2982,58 @@ type PodSpec struct {
 	// Optional: Defaults to true.
 	// +optional
 	EnableServiceLinks *bool `json:"enableServiceLinks,omitempty" protobuf:"varint,30,opt,name=enableServiceLinks"`
+	// TopologySpreadConstraints describes how a group of pods ought to spread across topology
+	// domains. Scheduler will schedule pods in a way which abides by the constraints.
+	// This field is alpha-level and is only honored by clusters that enables the EvenPodsSpread
+	// feature.
+	// +optional
+	// +patchMergeKey=topologyKey
```

Comment:
According to your validation code, there's actually two keys. Please set the list keys as here: https://github.com/kubernetes/api/blob/kubernetes-1.15.0-alpha.0/core/v1/types.go#L2082-L2095

Reply:
ah we are missing this 
```
listType=map
	// +listMapKey=containerPort
	// +listMapKey=protocol
``` , wondering how is this used ?

Reply:
Done. Nice to know this "black magic".